File: backend/app/schemas/comparison.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class PlatformPrice(BaseModel):
    platform: str
    url: str
    name: str
    current_price: float
    original_price: float | None
    discount_pct: float | None
    in_stock: bool
    image_url: str | None
    avg_rating: float | None
    is_source: bool


class ComparisonResult(BaseModel):
    product_id: uuid.UUID
    compared_at: datetime
    results: list[PlatformPrice]
    cheapest_platform: str
    price_diff: float
    price_diff_pct: float

File: frontend/src/types/comparison.ts
export interface PlatformPrice {
  platform: string;
  url: string;
  name: string;
  current_price: number;
  original_price: number | null;
  discount_pct: number | null;
  in_stock: boolean;
  image_url: string | null;
  avg_rating: number | null;
  is_source: boolean;
}

export interface ComparisonResult {
  product_id: string;
  compared_at: string;
  results: PlatformPrice[];
  cheapest_platform: string;
  price_diff: number;
  price_diff_pct: number;
}
